required/latex-lab/latex-lab-graphic.dtx

```diff
@@ -137,37 +384,211 @@
-%  structure but in a \texttt{Span} structure and no \texttt{/BBox} attribute is added.
-%  This in accordance with (the draft of) PDF/UA-2 but violates perhaps PDF/UA-1.
 %  \item[\texttt{correct-BBox}] If the calculated \texttt{/BBox} values are wrong
-%  they can be correct with this key. It expects four dimensions that are added to the 
+%  they can be corrected with this key. It expects four dimensions that are added to the 
 %  \texttt{/BBox} values. 
-%  \item[\texttt{debug}] The value \texttt{BBox} will show the calculated  \texttt{/BBox} 
-%  as a half transparent red rectangle. 
 %  \end{description}
 %  
-%  The code also redefines the \texttt{alt} key to actually add its values
-%  as an alternative text. If no \texttt{alt} value is given, a warning is issued 
-%  and the file name of the graphic is used.
+%  The code also add to the \texttt{debug} key of \cs{DocumentMetadata} the value \texttt{BBox}.
+%  This adds a half transparent red layer showing the calculated BBox.
+%  The key can only be used after that graphic module has been loaded, which means
+%  one needs a second \cs{DocumentMetadata} call:
+%  \begin{verbatim}
+%   \DocumentMetadata{tagging=on}
+%   \DocumentMetadata{debug=BBox}
+%  \end{verbatim}
+%  
+%  \section{Make \env{picture} tagging aware}
+%  
+%  \subsection{User interface}
+%  The \env{picture} environment doesn't have a optional argument with key-value value processing.
+%  So the code changes that and then provides the keys already discussed for \cs{includegraphics}.
+%  
+%  \subsection{BBox calculation}
+%  
+%  The BBox calculation is much simpler than for \cs{includegraphics} as the code simply takes
+%  the declared size from the \env{picture} arguments. 
+%  
+%  \subsection{Tagging in text mode}
 %  
+%  A  \env{picture} can contain \cs{put} commands with text content, tagging this in 
+%  text mode could make sense in some cases. But it is not quite clear if one can/should 
+%  redefine the \cs{put} command. If text tagging is wanted we suggest to 
+%  define a dedicated command along these lines:
+%  \begin{verbatim}
+%  \NewDocumentCommand\picturenode{O{}r()m}
+%    {
+%     \group_begin:
+%     \keys_set:nn{tag/graphic}{#1}
+%     \str_if_eq:VnT\l_tag_graphic_mode_tl {text}
+%      {\tag_resume:n{\picturenode}}
+%     \tag_socket_use:n{graphic/text/begin}
+%     \put(#2){#3}
+%     \tag_socket_use:n{graphic/text/end}  
+%     \group_end:
+%    }   
+%  
+%  \end{verbatim}   
+%  
+%  \section{Make an l3draw environment tagging aware}
+%  The newest l3draw version (in the latex3 github)
+%  has all needed data to define a command to retrieve the BBox
+%  and to build a tagging aware environment. l3draw currently has
+%  no dedicated function to add text, instead one has to store the text
+%  in a box and then reuse it, so similar to the \env{picture} environment
+%  a dedicated command with tagging awareness is suggested.
```

Comment:
same argument as before: there should be picture elements for text

Reply:
yes, here I naturally hope that @josephwright reads that and adds proper support ;-)

Reply:
I've not got around to emulated nodes ... they are complicated! (The idea was that simple text isn't really a drawing thing at all - it's simply a box used inside a picture).